File: src/main/java/alfheim/client/render/block/RenderBlockItemHolder.java
package alfheim.client.render.block;

import static org.lwjgl.opengl.GL11.*;

import alfheim.api.lib.LibRenderIDs;
import alfheim.common.block.tile.TileItemHolder;
import cpw.mods.fml.client.registry.ISimpleBlockRenderingHandler;
import net.minecraft.block.Block;
import net.minecraft.client.renderer.RenderBlocks;
import net.minecraft.client.renderer.tileentity.TileEntityRendererDispatcher;
import net.minecraft.world.IBlockAccess;

public class RenderBlockItemHolder implements ISimpleBlockRenderingHandler {

	@Override
	public void renderInventoryBlock(Block block, int metadata, int modelID, RenderBlocks renderer) {
		glPushMatrix();
		glTranslated(-0.5, 0.25, -0.5);
		TileEntityRendererDispatcher.instance.renderTileEntityAt(new TileItemHolder(), 0, 0, 0, 0);
		glPopMatrix();
	}

	@Override
	public boolean renderWorldBlock(IBlockAccess world, int x, int y, int z, Block block, int modelId, RenderBlocks renderer) {
		return false;
	}

	@Override
	public int getRenderId() {
		return LibRenderIDs.idItemHolder;
	}

	@Override
	public boolean shouldRender3DInInventory(int modelId) {
		return true;
	}
}

File: src/main/java/alfheim/client/render/block/RenderBlockItemHolder.kt
package alfheim.client.render.block

import org.lwjgl.opengl.GL11.*

import alfheim.api.lib.LibRenderIDs
import alfheim.common.block.tile.TileItemHolder
import cpw.mods.fml.client.registry.ISimpleBlockRenderingHandler
import net.minecraft.block.Block
import net.minecraft.client.renderer.RenderBlocks
import net.minecraft.client.renderer.tileentity.TileEntityRendererDispatcher
import net.minecraft.world.IBlockAccess

class RenderBlockItemHolder: ISimpleBlockRenderingHandler {
	
	override fun renderInventoryBlock(block: Block, metadata: Int, modelID: Int, renderer: RenderBlocks) {
		glPushMatrix()
		glTranslated(-0.5, 0.25, -0.5)
		TileEntityRendererDispatcher.instance.renderTileEntityAt(TileItemHolder(), 0.0, 0.0, 0.0, 0f)
		glPopMatrix()
	}
	
	override fun renderWorldBlock(world: IBlockAccess, x: Int, y: Int, z: Int, block: Block, modelId: Int, renderer: RenderBlocks): Boolean {
		return false
	}
	
	override fun getRenderId(): Int {
		return LibRenderIDs.idItemHolder
	}
	
	override fun shouldRender3DInInventory(modelId: Int): Boolean {
		return true
	}
}
